### DISCUSSION TITLE
How to query the broker resources

### DISCUSSION THREAD

**Discussion by hrohmer** [Mobility-Data-Space/mobility-data-space#6] [General]
Hello all,

we'd like to query the broker right from our DataspaceConnector to request the offers of all (or certain) parties. We'd like to use the DSC API /admin/api/broker/query and send a SPARQL request.

Is there any example SPARQL query? Right now we don't know how to fill up the triple to get the offers.

**Comment by JohannesLipp** [comment]
Are you looking for offered contracts or resources? I am not familiar with the DSC architecture, but you might be able to extend the following example to fulfill your needs:

```
PREFIX ids: <https://w3id.org/idsa/core/>


SELECT ?resource ?offer
WHERE {
  ?resource ids:contractOffer ?offer .
}
LIMIT 25
```
